Signal/src/views/OWSFlatButton.swift

```diff
@@ -0,0 +1,176 @@
+    }
+
+    private func createContent() {
+        let button = UIButton(type:.custom)
+        self.button = button
+        self.addSubview(button)
+        button.addTarget(self, action:#selector(buttonPressed), for:.touchUpInside)
+        button.autoPinWidthToSuperview()
+        button.autoPinHeightToSuperview()
+    }
+
+    public class func button(title: String,
+                             font: UIFont,
+                             titleColor: UIColor,
+                             backgroundColor: UIColor,
+                             width: CGFloat,
+                             height: CGFloat,
+                             target:Any,
+                             selector: Selector) -> OWSFlatButton {
+        let button = OWSFlatButton()
+        button.setTitle(title:title,
+                        font: font,
+                        titleColor: titleColor )
+        button.setBackgroundColors(backgroundColor)
+        button.useDefaultCornerRadius()
+        button.setSize(width:width, height:height)
+        button.addTarget(target:target, selector:selector)
+        return button
+    }
+
+    public class func button(title: String,
+                             titleColor: UIColor,
+                             backgroundColor: UIColor,
+                             width: CGFloat,
+                             height: CGFloat,
+                             target:Any,
+                             selector: Selector) -> OWSFlatButton {
+        return OWSFlatButton.button(title:title,
+                                    font:fontForHeight(height),
+                                    titleColor:titleColor,
+                                    backgroundColor:backgroundColor,
+                                    width:width,
+                                    height:height,
+                                    target:target,
+                                    selector:selector)
+    }
+
+    public class func button(title: String,
+                             font: UIFont,
+                             titleColor: UIColor,
+                             backgroundColor: UIColor,
+                             target:Any,
+                             selector: Selector) -> OWSFlatButton {
+        let button = OWSFlatButton()
+        button.setTitle(title:title,
+                        font: font,
+                        titleColor: titleColor )
+        button.setBackgroundColors(backgroundColor)
+        button.useDefaultCornerRadius()
+        button.addTarget(target:target, selector:selector)
+        return button
+    }
+
+    public class func fontForHeight(_ height: CGFloat) -> UIFont {
+        let fontPointSize = round(height * 0.45)
+        return UIFont.ows_mediumFont(withSize:fontPointSize)!
+    }
+
+    // MARK: Methods
+
+    public func setTitle(title: String, font: UIFont,
+                         titleColor: UIColor ) {
+        guard let button = self.button else {
```

Comment:
nit: Is there a reason `self.button` isn't assigned in init so we can avoid these guards?

Reply:
Done.  I wanted to gather "creation & configuration of views" logic in `createContent()` but pragmatism > purity.